File: http patch/Scala/fundservice/fund/src/main/scala/fund/PatchInstructionService.scala
package fund

import fund.DomainModel._

object PatchInstructionService {

  def patchFundType(fund: Fund, instruction: PatchFundDto) = {
    instruction match {
      case instruction if instruction.operation == "replace" => updateFundType(instruction.value)
    }
  }

  def patchFundName(fund: Fund, instruction: PatchFundDto) = {
    instruction match {
      case instruction if instruction.operation == "replace" => updateFundName(instruction.value)
    }
  }
}

File: http patch/C#/FundService/FundService/PatchInstructionService.cs
using FundService.Common;
using System;
using static FundService.DomainModel;

namespace FundService
{
    public static class PatchInstructionService
    {
        public static Func<Fund, Result<Fund>> PatchFundType(Fund fund, PatchFundDto instruction)  {
            switch (instruction.Operation) {
                case "replace": return UpdateFundType(instruction.Value);
            }
            return null;
        }
        public static Func<Fund, Result<Fund>> PatchFundName(Fund fund, PatchFundDto instruction)
        {
            switch (instruction.Operation) {
                case "replace": return UpdateFundName(instruction.Value);
            }
            return null;
        }
    }
    
}
